### ISSUE TITLE
NavMeshGenerator.java uses Thread.stop()

### ISSUE THREAD

**Issue by tonihele** [jMonkeyEngine/sdk#217]
NavMeshGenerator.java uses Thread.stop(). There is a timeout safeguard that kills of the NavMesh generator thread with this function. Basically changing it to use interrupt() wont work as intended. The NavMesh generator would need to support checking for the interruptions.

Decide what to do.

**Comment by MeFisto94** [comment]
To quote @pspeed42 on that:

TLDR is:
> If there is some library that we cannot control and feel the need to Thread.stop() it then it should be spun off into its own process that we can kill.
Extra complexity for transferring data back-and-forth, etc.
But Thread.stop() is a no-no.

Longer Version:
> re: Thread.stop()... the only way to safely murder a thread in Java is System.exit().  Thread.stop() may leave the application in a weird state because it screws with the order of finalizers being called, etc... bad ju-ju.  Applications that really MUST call Thread.stop() should only do it as a precursor to killing themselves (System.exit()).
One should never call Thread.stop() as a normal function of thread management.  It is better to ask threads to commit suicide.  Thread.interrupt() is not necessary unless your thread is pausing from blocking IO or something inside Java.  Generally, a thread-safe boolean is enough (AtomicBoolean or volatile boolean).
If the thing you are calling is not iterative in nature then you are sort of screwed.
But recognize that calling Thread.stop() is only trading one problem for other potential random problems.
There are some methods in Java where it should be required to put on safety gear first... and Thread.stop() is one of them.

I think we're better of looking into improving the NavMeshGenerator Thread. Depending on the way this works, we might also look into https://github.com/MeFisto94/jme3-recast4j and especially tiled builds, where we could just stop the executor from building further tiles.

**Comment by MeFisto94** [comment]
The Thread is basically run from https://github.com/jMonkeyEngine/sdk/blob/e9d4b2a8bd4e4822e30d68a0f076ee14a7c12bdc/jme3-navmesh-gen/src/com/jme3/gde/nmgen/NavMeshGenerator.java#L527 so we don't have a real choice to interrupt the Thread
